src/app/modules/shared_models/link_preview_model.nim

```diff
@@ -276,3 +290,67 @@ QtObject:
     result = @[]
     for item in self.items:
       result.add(item.linkPreview.url)
+
+
+  proc getContactIds*(self: Model): HashSet[string] =
+    for item in self.items:
+      let contactId = item.linkPreview.getContactId()
+      if contactId != "":
+        result.incl(contactId)
+
+  proc getCommunityIds*(self: Model): HashSet[string] =
+    for item in self.items:
+      let communityId = item.linkPreview.getCommunityId()
+      if communityId != "":
+        result.incl(communityId)
+
+  proc setItemLoadingLocalData(self: Model, row: int, item: Item, value: bool) = 
+    if item.loadingLocalData == value:
+      return
+    item.loadingLocalData = value
+    let modelIndex = self.createIndex(row, 0, nil)
+    defer: modelIndex.delete
+    self.dataChanged(modelIndex, modelIndex, @[ModelRole.LoadingLocalData.int])
+
+  proc setItemIsLocalData(self: Model, row: int, item: Item) = 
+    if item.isLocalData:
+      return
+    var roles = @[ModelRole.IsLocalData.int]
+    item.isLocalData = true
+    if item.loadingLocalData:
+      item.loadingLocalData = false
+      roles.add(ModelRole.LoadingLocalData.int)
+    let modelIndex = self.createIndex(row, 0, nil)
+    defer: modelIndex.delete
+    self.dataChanged(modelIndex, modelIndex, roles)
+
+  proc setContactInfo*(self: Model, contactDetails: ContactDetails) =
+    for row, item in self.items:
+      if item.linkPreview.setContactInfo(contactDetails):
+        self.setItemIsLocalData(row, item)
+
+  proc setCommunityInfo*(self: Model, community: CommunityDto) =
+    for row, item in self.items:
```

Comment:
Just suggestion: replace seq 'self.items' with table<string(id), Item>

Reply:
Hmm, that's not really possible, since it's a `Model`.  Items here could be added and removed, we will have to track this and update the map.

And there's not many items here, so should be fine as is. 
But we can try to improve later if we want to 🙂